File: tests/fixtures/sample.go
package main

import (
	"fmt"
	"strings"
)

// UserService manages user operations.
type UserService struct {
	DBPath string
	limit  int
}

// Greeter interface for greeting.
type Greeter interface {
	Greet(name string) string
}

// GetUser retrieves a user by ID.
func (s *UserService) GetUser(userID int) (map[string]interface{}, error) {
	return nil, nil
}

func (s *UserService) validate(data map[string]interface{}) bool {
	return true
}

// CreateService is a factory function.
func CreateService(path string) *UserService {
	return &UserService{DBPath: path}
}

func helper() {
	fmt.Println("internal")
	_ = strings.TrimSpace("")
}

const MaxRetries = 3

File: tests/fixtures/sample.rs
use std::collections::HashMap;
use std::fmt;

pub struct UserService {
    pub db_path: String,
    limit: i32,
}

pub trait Greeter {
    fn greet(&self, name: &str) -> String;
}

impl UserService {
    pub fn new(db_path: String) -> Self {
        UserService { db_path, limit: 10 }
    }

    pub fn get_user(&self, user_id: i32) -> Option<HashMap<String, String>> {
        None
    }

    fn validate(&self, data: &str) -> bool {
        true
    }
}

impl Greeter for UserService {
    fn greet(&self, name: &str) -> String {
        format!("Hello, {}", name)
    }
}

pub fn create_service(path: String) -> UserService {
    UserService::new(path)
}

fn helper() -> bool {
    true
}

pub const MAX_RETRIES: i32 = 3;

pub enum Status {
    Active,
    Inactive,
    Pending,
}
